### PR TITLE
Update functions calls to be delegated

### LINKED ISSUE #128: Encapsulate functions to use proxys
Some functions as https://github.com/alastria/alastria-identity-lib/blob/fa6a3d9a1b2acce3c2ed815bf584a8455cff059a/src/txFactory/identityManagerTransactionFactory.ts#L66-L76 needs to go delegated (go through the proxy).

In order to archieve that, the `transaction.data` needs to be like `delegated(web3, encapsulatedData)` where `encapsulatedData` is the `encodeFunctionCall`.

**Check all functions to know which ones need to be fixed!**
Already identified:
- [x] addIdentityIssuer
- [x] updateIdentityIssuerEidasLevel
- [x] deleteIdentityIssuer
- [x] addIdentityServiceProvider
- [x] deleteIdentityServiceProvider
- [x] All entity related

### PR DESCRIPTION
**Pull request by DrankoLQ**
close #128 

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Documentation

## Description
This PR updates some functions calls to be delegated so they can go through the proxy

### CODE REVIEW

**Comment by DrankoLQ** [comment]
A question @VictorNS69: is it supposed to fail some examples with this change? Correct me if I'm wrong, but the delegated calls is used with upgradeable smart contracts only and we don't have them yet, isn't it?

**Comment by VictorNS69** [comment]
> A question @VictorNS69: is it supposed to fail some examples with this change? Correct me if I'm wrong, but the delegated calls are used with upgradeable smart contracts only and we don't have them yet, isn't it?

1) Some examples will fail if you use them in the wrong order. For example, if you try to delete an Issuer that does not have an issuer role. However, if the error is something like "invalid RPC call", "invalid arguments", etc it means something is not working. **If you have any specific errors, just show me the log**
2) **Delegated calls are used in both contracts**, upgradeable and non-upgradeable. **You should test with the non-upgradeable version (master)** because the upgradeable version is not working at the moment. See #116

**Comment by VictorNS69** [comment]
@DrankoLQ I recomend you to print `delegatedData` and `transaction.data`. The values should be close to `0x889776a80000000000000000000000004b750a9a228afbcc97aa9e0ac967b0da9aab2f1b0000000000000000000000000000000000000000000000000000000000000003` for `delegatedData` and `0x597b2e9b00000000000000000000000061087480e211aefa69a103a28a34844a41eeb380000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000044889776a80000000000000000000000004b750a9a228afbcc97aa9e0ac967b0da9aab2f1b000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000` for `transaction.data`.

**At least the begining of each hex should be the same I think.**

**Comment by DrankoLQ** [comment]
I have tested the example in the right order and everything works @VictorNS69  🙂

File: src/txFactory/identityManagerTransactionFactory.ts [hunk 1]
```diff
@@ -327,10 +338,11 @@ export function setUrlAOA(web3, didEntity, urlAOA) {
 export function getEntity(web3, didEntity) {
   const entityAddr = AIdUtils.getProxyAddress(didEntity)
   const transaction = Object.assign({}, config.basicTransaction)
-  transaction.data = web3.eth.abi.encodeFunctionCall(
+  const delegatedData = web3.eth.abi.encodeFunctionCall(
     config.contractsAbi.AlastriaIdentityEntity.getEntity,
     [entityAddr]
   )
+  transaction.data = delegated(web3, delegatedData)
```
Comment (VictorNS69) [inline src/txFactory/identityManagerTransactionFactory.ts:345]: In a call function (getters) there is no need to use the proxy. Revert this function.

File: src/txFactory/identityManagerTransactionFactory.ts [hunk 2]
```diff
@@ -342,10 +354,11 @@ export function getEntity(web3, didEntity) {
  */
 export function entitiesList(web3) {
   const transaction = Object.assign({}, config.basicTransaction)
-  transaction.data = web3.eth.abi.encodeFunctionCall(
+  const delegatedData = web3.eth.abi.encodeFunctionCall(
     config.contractsAbi.AlastriaIdentityEntity.entitiesList,
     []
   )
+  transaction.data = delegated(web3, delegatedData)
```
Comment (VictorNS69) [inline src/txFactory/identityManagerTransactionFactory.ts:361]: In a call function (getters) there is no need to use the proxy. Revert this function.

**Review by VictorNS69 — CHANGES_REQUESTED** [review]

**Comment by VictorNS69** [comment]
The rest is working properly

**Review by VictorNS69 — APPROVED** [review]

**Pull request merged**